File: surpass/surpass-commons/surpass-commons-common/src/main/java/org/dromara/surpass/enums/VoucherStatusEnum.java
/*
 * Copyright [2025] [JinBooks of copyright http://www.jinbooks.com]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 */


package org.dromara.surpass.enums;

import lombok.Getter;

import java.util.LinkedHashMap;
import java.util.Map;

/**
 * 记账凭证状态
 */
@Getter
public enum VoucherStatusEnum implements BaseEnum {
    DRAFT("draft", "暂存"),
    UNDER_REVIEW("reviewing", "审核中"),
    COMPLETED("completed", "已完成"),
    REJECTED("rejected", "被拒绝"),
    CANCELLED("cancelled", "已取消");

    private final String value;
    private final String label;

    VoucherStatusEnum(String value, String label) {
        this.value = value;
        this.label = label;
    }

    public static VoucherStatusEnum get(String value) {
        for (VoucherStatusEnum status : VoucherStatusEnum.values()) {
            if (status.getValue().equals(value)) {
                return status;
            }
        }
        return null;
    }

    @Override
    public Map<Object, Object> getMap() {
        Map<Object, Object> map = new LinkedHashMap<>();
        for (VoucherStatusEnum status : VoucherStatusEnum.values()) {
            map.put(status.getValue(), status.getLabel());
        }
        return map;
    }
}

File: surpass-ui/src/utils/enums/VoucherStatusEnum.ts
enum Status {
    DRAFT = "draft",            // 暂存
    UNDER_REVIEW = "reviewing", // 审核中
    COMPLETED = "completed",    // 已完成
    REJECTED = "rejected",      // 被拒绝
    CANCELLED = "cancelled"     // 已取消
}

// 中文描述的映射对象
const statusDescriptions: { [key in Status]: string } = {
    [Status.DRAFT]: "暂存",
    [Status.UNDER_REVIEW]: "审核中",
    [Status.COMPLETED]: "已完成",
    [Status.REJECTED]: "被拒绝",
    [Status.CANCELLED]: "已取消"
};

// 对应状态的颜色映射
const statusColors: { [key in Status]: string } = {
    [Status.DRAFT]: "#909399",        // 暂存
    [Status.UNDER_REVIEW]: "#E6A23C", // 审核中
    [Status.COMPLETED]: "#67C23A",    // 已完成
    [Status.REJECTED]: "#F56C6C",     // 被拒绝
    [Status.CANCELLED]: "#909399"     // 已取消
};

// 根据输入值获取中文描述，并添加颜色
export function getVoucherStatusDesc(value: string): string {
    // 使用枚举值反向查找映射
    const status = Object.values(Status).find(status => status === value);
    if (status) {
        // 获取对应状态的颜色
        const color = statusColors[status];
        // 返回带有颜色的 span 标签
        return `<span style="color: ${color}">${statusDescriptions[status]}</span>`;
    }
    return value;
}
